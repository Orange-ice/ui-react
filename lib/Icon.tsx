import React from 'react';

interface IconProps {
  name: string
}

const Icon:React.FunctionComponent<IconProps> = (props) => {
  return (
    <span>Icon---{props.name}</span>
  )
}

export default Icon
