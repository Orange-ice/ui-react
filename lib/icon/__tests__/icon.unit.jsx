import React from 'react'
import Icon from '../icon'
import rederer from 'react-test-renderer'
import {mount} from 'enzyme'

describe('Icon 组件测试', () => {
  it('渲染', function () {
    const dom = rederer.create(<Icon name={'github'}/>).toJSON()
    expect(dom).toMatchSnapshot()
  })
  it('click事件', () => {
    const fn = jest.fn()
    const component = mount(<Icon name={'github'} onClick={fn}/>)
    component.find('svg').simulate('click')
    expect(fn).toBeCalled()
  })
})
