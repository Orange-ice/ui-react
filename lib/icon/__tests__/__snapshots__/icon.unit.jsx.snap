// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Icon 组件测试 渲染 1`] = `
<svg
  className="ice-icon"
>
  <use
    xlinkHref="#github"
  />
</svg>
`;
